lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/openxjson/OpenXJsonDeserializer.java

```diff
@@ -288,14 +290,11 @@ void decodeValue(Object jsonValue, BlockBuilder builder)
 
             try {
                 long longValue = parseLong(jsonString.value());
-                if ((int) longValue == longValue) {
-                    INTEGER.writeLong(builder, longValue);
-                    return;
-                }
+                INTEGER.writeLong(builder, longValue);
```

Comment:
On integer overflow (`(int) longValue != longValue`), `IntegerType#writeLong` will throw a `TrinoException(GENERIC_INTERNAL_ERROR, ...)` instead of `TrinoException(BAD_DATA, ...)`. I think we should check for that explicitly and throw `BAD_DATA` on both, no?